src/services/watchFolder/watchFolderService.ts

```diff
@@ -0,0 +1,512 @@
+        const notDeletedMappings = [];
+        for (const mapping of mappings) {
+            const mappingExists = await this.ElectronAPIs.doesFolderExists(
+                mapping.folderPath
+            );
+            if (mappingExists) {
+                notDeletedMappings.push(mapping);
+            }
+        }
+        this.ElectronAPIs.setWatchMappings(notDeletedMappings);
+        return notDeletedMappings;
+    }
+
+    async pushEvent(event: EventQueueItem) {
+        this.eventQueue.push(event);
+        debounce(this.runNextEvent.bind(this), 300)();
+    }
+
+    async pushTrashedDir(path: string) {
+        this.trashingDirQueue.push(path);
+    }
+
+    private setWatchFunctions() {
+        if (this.allElectronAPIsExist) {
+            this.ElectronAPIs.registerWatcherFunctions(
+                diskFileAddedCallback,
+                diskFileRemovedCallback,
+                diskFolderRemovedCallback
+            );
+        }
+    }
+
+    async addWatchMapping(
+        collectionName: string,
+        folderPath: string,
+        hasMultipleFolders: boolean
+    ) {
+        if (this.allElectronAPIsExist) {
+            try {
+                await this.ElectronAPIs.addWatchMapping(
+                    collectionName,
+                    folderPath,
+                    hasMultipleFolders
+                );
+            } catch (e) {
+                logError(e, 'error while adding watch mapping');
+            }
+        }
+    }
+
+    async removeWatchMapping(folderPath: string) {
+        if (this.allElectronAPIsExist) {
+            try {
+                await this.ElectronAPIs.removeWatchMapping(folderPath);
+            } catch (e) {
+                logError(e, 'error while removing watch mapping');
+            }
+        }
+    }
+
+    getWatchMappings(): WatchMapping[] {
+        if (this.allElectronAPIsExist) {
+            try {
+                return this.ElectronAPIs.getWatchMappings() ?? [];
+            } catch (e) {
+                logError(e, 'error while getting watch mappings');
+                return [];
+            }
+        }
+        return [];
+    }
+
+    private setIsEventRunning(isEventRunning: boolean) {
+        this.isEventRunning = isEventRunning;
+        this.setWatchFolderServiceIsRunning(isEventRunning);
+    }
+
+    private async runNextEvent() {
+        console.log('mappings', this.getWatchMappings());
+
+        if (this.eventQueue.length === 0 || this.isEventRunning) {
+            return;
+        }
+
+        this.setIsEventRunning(true);
+        const event = this.clubSameCollectionEvents();
+        this.currentEvent = event;
+        console.log('running event', event);
+        if (event.type === 'upload') {
+            this.processUploadEvent();
+        } else {
+            await this.processTrashEvent();
+            this.setIsEventRunning(false);
+            this.runNextEvent();
+        }
+    }
+
+    private async processUploadEvent() {
+        try {
```

Comment:
what will happen, lets a desktopUpload resume starts on app start,this will then override that na ?

we need to have some upload already running state and check that before running this